src/Project/functions/copyItem.ts

```diff
@@ -1,12 +1,17 @@
 import fs from "fs-extra";
 import { ProjectData } from "Project";
+import { checkDeclarationFileFilename } from "Project/functions/checkFileName";
 import { isCompilableFile } from "Project/util/isCompilableFile";
 import { PathTranslator } from "Shared/classes/PathTranslator";
 import { DTS_EXT } from "Shared/constants";
 
 export function copyItem(data: ProjectData, pathTranslator: PathTranslator, item: string) {
 	fs.copySync(item, pathTranslator.getOutputPath(item), {
 		filter: (src, dest) => {
+			if (src.endsWith(DTS_EXT)) {
+				checkDeclarationFileFilename(src);
+			}
```

Comment:
...This is probably okay for now, but I am very uncertain about putting this logic in the filter function.